### PR TITLE
Revert changes for course

### PR DESCRIPTION
**Pull request by awais786**
Revert all changes to a previous version.
https://openedx.atlassian.net/browse/ECOM-7776

By default button will not appear.
screen shot 2017-05-08 at 12 53 07 pm

-----------------------------

If history version is not current one then this button will appear.
screen shot 2017-05-08 at 12 52 51 pm

-----------------------------

Confirmation Box added to make sure revert does not occur mistakenly.
screen shot 2017-05-08 at 1 56 00 pm

### CODE REVIEW

**Comment by awais786** [comment]
@tasawernawaz @waheedahmed Please review.

File: course_discovery/apps/publisher/api/views.py [hunk 1]
```diff
@@ -28,6 +38,8 @@ def get_queryset(self):
         queryset = User.objects.filter(groups__name=org_extension.group).order_by('full_name', 'username')
         return queryset
 
+    def patch(self):
+        pass
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/api/views.py]: remove this unused code.

File: course_discovery/apps/publisher/api/views.py [hunk 2]
```diff
@@ -36,7 +48,7 @@ class UpdateCourseRunView(UpdateAPIView):
     serializer_class = CourseRunSerializer
 
 
-class CourseRevisionDetailView(RetrieveAPIView):
+class CourseRevisionDetailView(RetrieveAPIView, UpdateAPIView):
```
Comment (tasawernawaz) [inline course_discovery/apps/publisher/api/views.py]: I think no need this.

**Review by tasawernawaz — APPROVED** [review]
Minor update required.

File: course_discovery/apps/publisher/api/views.py [hunk 3]
```diff
@@ -56,3 +66,24 @@ class ChangeCourseRunStateView(UpdateAPIView):
...
     serializer_class = CourseRunStateSerializer
+
+
+class RevertCourseReversionView(APIView):
+    """ Revert view for Course against a history version """
+    permission_classes = (IsAuthenticated, )
+
+    def put(self, request, history_id):  # pylint: disable=unused-argument
+        """ Update the course version against the given revision id. """
+        history = get_object_or_404(historicalcourse, pk=history_id)
```
Comment (waheedahmed) [inline course_discovery/apps/publisher/api/views.py]: `history_object`

File: course_discovery/apps/publisher/api/urls.py [hunk 4]
```diff
@@ -14,4 +14,5 @@
     url(r'^course_runs/(?P<pk>\d+)/$', UpdateCourseRunView.as_view(), name='update_course_run'),
     url(r'^course_revisions/(?P<history_id>\d+)/$', CourseRevisionDetailView.as_view(), name='course_revisions'),
     url(r'^course_run_state/(?P<pk>\d+)/$', ChangeCourseRunStateView.as_view(), name='change_course_run_state'),
+    url(r'^course/revision/(?P<history_id>\d+)/revert/$', RevertCourseReversionView.as_view(), name='change_revision_revert'),
```
Comment (waheedahmed) [inline course_discovery/apps/publisher/api/urls.py]: Change the url name to more appropriate name.

File: course_discovery/apps/publisher/api/views.py [hunk 5]
```diff
@@ -56,3 +66,24 @@ class ChangeCourseRunStateView(UpdateAPIView):
     permission_classes = (IsAuthenticated, PublisherUserPermission,)
     queryset = CourseRunState.objects.all()
     serializer_class = CourseRunStateSerializer
+
+
+class RevertCourseReversionView(APIView):
```
Comment (waheedahmed) [inline course_discovery/apps/publisher/api/views.py]: Same here, change the view name.

**Review by waheedahmed — CHANGES_REQUESTED** [review]

**Review by waheedahmed — APPROVED** [review]

**Pull request merged**